### PR TITLE
Remove single file storage

### PR DESCRIPTION
**Pull request by harlem88**
- Introduce a directory for data storage so that all data are confined to the same location and do not have files scattered throughout the system.
- Check configured path before device-runtime starts.
- Add some tests on get_credentials function.

### CODE REVIEW

**Review by rbino — CHANGES_REQUESTED** [review]
Additional note: this would break compatibility for existing devices with credentials saved already saved in the store file, they would have to be manually unregistered. Is this acceptable? @bettio

File: src/lib.rs [hunk 1]
```diff
@@ -65,7 +65,15 @@ pub struct DeviceManager {
 impl DeviceManager {
     pub async fn new(opts: DeviceManagerOptions) -> Result<DeviceManager, DeviceManagerError> {
         let device_id: String = get_device_id(opts.device_id.clone()).await?;
-        let credential_secret: String = get_credentials_secret(&device_id, &opts).await?;
+        let credential_secret: String = get_credentials_secret(
```
Comment (Annopaolo) [inline src/lib.rs:68]: Using `credentials_secret` as variable name would be more consistent

File: src/lib.rs [hunk 2]
```diff
@@ -65,7 +65,15 @@ pub struct DeviceManager {
...
+            &device_id,
+            &opts,
+            FileStateRepository::new(
+                opts.store_directory.clone(),
+                format!("credential_{}.json", device_id),
```
Comment (Annopaolo) [inline src/lib.rs:73]: See above, s/credential/credentials

**Review by Annopaolo — CHANGES_REQUESTED** [review]
Just a minor comment

**Review by Annopaolo — APPROVED** [review]

**Review by rbino — APPROVED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**